src/main/java/frc/robot/subsystems/gripper/Gripper.java

```diff
@@ -51,8 +54,14 @@ public boolean atSetpoint() {
                 inputs.angleSetpoint, GripperConstants.THRESHOLD.in(Units.Percent));
     }
 
+    public boolean isReversed() {
+        return inputs.currentAngle.isNear(
+                GripperConstants.WRIST_BACKWARDS_AMP,
+                GripperConstants.WRIST_TOLERANCE.in(Units.Rotations));
+    }
+
     public Command setRollerPower(double power) {
-        return runOnce(() -> io.setRollerMotorPower(power)).withName("set roller power");
+        return Commands.runOnce(() -> io.setRollerMotorPower(power)).withName("set roller power");
```

Comment:
If this command doesn't have a requirement, it should be private. Otherwise it may be used incorrectly.

Reply:
3d00846

Reply:
I think the same should be done in the gripper's code.

Reply:
No, the setRollerPower & setRollerSpeed is used in other places

Reply:
That means that you run commands without requirements which is bad. The code needs to be refactored a bit to avoid these situations.
BTW, this is one of the advantages of splitting subsystems farther than how they are now 🙃

Reply:
e77aa3930ea59b634ac19b6906610d1780769904

Reply:
Well with this change you can use `Subsystem's` `runOnce()`. I believe there was a reason to remove the requirement. Probably to avoid it canceling commands of different parts of the same subsystem.

Reply:
I think it'll be beneficial to pass over all of the commands involving more than one part of a subsystem and check their requirements. Remember that command groups may also be constructed inside the subsystem's class.

Reply:
We checked all of the command groups
There are no command groups within the subsystem

> you can use Subsystem's runOnce()

The command is already using `runOnce()`.

Reply:
> There are no command groups within the subsystem

Let me rephrase -- I meant that you can create command groups inside a subsystem if that'll help manage requirements.
> The command is already using `runOnce()`.

It uses `Commands.runOnce()`, not `Subsystem.runOnce()` which was what I suggested. If you're sure `setRollerPower()` needs to require the subsystem just use the latter.

Reply:
>If you're sure setRollerPower() needs to require the subsystem

You were right it can't require the gripper but we're not going to split the gripper now so I'm reverting the commit.

Reply:
c0db7c40db35c7356a9f2efcbfb10515a4d79c84

Reply:
I don't suggest splitting the gripper _now_ (this is just a lesson for next year). My suggestion is in the first part of my latest reply.

Reply:
e7a1fef

Reply:
The `setRollerPower` command is still public and doesn't require the subsystem.

Reply:
I don't see how setRollerCommand can be private because it is used in other places.
If you have a better solution feel free to suggest it
349daa0

Reply:
> I don't suggest splitting the gripper _now_ (this is just a lesson for next year). My suggestion is to create private commands without requirements, then a public command composition that uses those private commands and does require the gripper.

https://github.com/Galaxia5987/robot-2024/pull/39/files/c58fe8093d099d1d76f37d6ab9373c8cc168c37d#r1485208017

Reply:
I don't see the problem in c580676

Reply:
There is no hyperlink to this commit so I don't think you pushed it

Reply:
I did push this commit

Reply:
Can you resolve this please?

Reply:
Pushed after writing your comment. Resolving